### ISSUE TITLE
[2.33] Exception reported by player

### ISSUE THREAD

**Issue by tchegito** [tchegito/zildo#108]
NullPointerException in DialogManagement

map=prison5 (62x37)
sprites=[Perso=Zildo
Coords:(731.75903, 437.00818 0.0)
info=ZILDO
mvt=VIDE, 732.75903, 447.00818 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 152.0, 159.0 (SHADOW), 747.75903, 458.00818 (103 - bank 0), 10.0, 333.0 (15 - bank 5), Entity id=509
x=644.0
y=32.0
nSpr=231
visible=true, Entity id=510
x=644.0
y=24.0
nSpr=232
visible=true, Entity id=511
x=652.0
y=112.0
nSpr=230
visible=true, Entity id=5
x=644.0
y=112.0
nSpr=232
visible=true, Entity id=6
x=676.0
y=112.0
nSpr=232
visible=true, Entity id=7
x=700.0
y=112.0
nSpr=232
visible=true, Entity id=8
x=700.0
y=112.0
nSpr=232
visible=true, Entity id=9
x=700.0
y=112.0
nSpr=232
visible=true, Entity id=10
x=708.0
y=112.0
nSpr=230
visible=true, Entity id=11
x=756.0
y=112.0
nSpr=232
visible=true, Entity id=12
x=756.0
y=112.0
nSpr=232
visible=true, Entity id=13
x=756.0
y=120.0
nSpr=231
visible=true, door3a
352.0, 80.0 (2 - bank 5), door3b
304.0, 80.0 (2 - bank 5), Entity id=16
x=284.0
y=96.0
nSpr=231
visible=true, Entity id=17
x=284.0
y=88.0
nSpr=232
visible=true, Entity id=18
x=372.0
y=88.0
nSpr=232
visible=true, Entity id=19
x=372.0
y=96.0
nSpr=231
visible=true, Entity id=20
x=756.0
y=216.0
nSpr=232
visible=true, Entity id=21
x=708.0
y=216.0
nSpr=230
visible=true, Entity id=22
x=700.0
y=216.0
nSpr=232
visible=true, Entity id=23
x=756.0
y=240.0
nSpr=232
visible=true, Entity id=24
x=756.0
y=248.0
nSpr=231
visible=true, Entity id=25
x=756.0
y=320.0
nSpr=232
visible=true, Entity id=27
x=764.0
y=320.0
nSpr=230
visible=true, Entity id=28
x=932.0
y=320.0
nSpr=232
visible=true, Entity id=29
x=692.0
y=240.0
nSpr=230
visible=true, Entity id=30
x=684.0
y=240.0
nSpr=232
visible=true, Entity id=31
x=932.0
y=328.0
nSpr=231
visible=true, Entity id=32
x=932.0
y=504.0
nSpr=232
visible=true, Entity id=33
x=660.0
y=504.0
nSpr=232
visible=true, Entity id=34
x=668.0
y=504.0
nSpr=230
visible=true, Entity id=35
x=692.0
y=504.0
nSpr=232
visible=true, Entity id=36
x=756.0
y=504.0
nSpr=232
visible=true, Entity id=37
x=764.0
y=504.0
nSpr=230
visible=true, Entity id=38
x=828.0
y=504.0
nSpr=232
visible=true, Entity id=39
x=908.0
y=504.0
nSpr=230
visible=true, Entity id=40
x=900.0
y=504.0
nSpr=232
visible=true, Entity id=41
x=548.0
y=216.0
nSpr=232
visible=true, Entity id=42
x=548.0
y=240.0
nSpr=232
visible=true, Entity id=43
x=556.0
y=240.0
nSpr=230
visible=true, Entity id=44
x=556.0
y=216.0
nSpr=230
visible=true, Entity id=45
x=660.0
y=240.0
nSpr=232
visible=true, Entity id=46
x=580.0
y=216.0
nSpr=232
visible=true, 302.0, 333.0 (STONE_HEAVY), 295.0, 322.0 (STONE_HEAVY), 291.0, 339.0 (STONE_HEAVY), 305.0, 317.0 (STONE_HEAVY), 313.0, 341.0 (STONE_HEAVY), 663.0, 527.0 (BARREL), doorc1b
736.0, 378.0 (16 - bank 5), doorc2b
848.0, 378.0 (15 - bank 5), Entity id=55
x=492.0
y=560.0
nSpr=232
visible=true, Entity id=56
x=492.0
y=544.0
nSpr=231
visible=true, Entity id=57
x=492.0
y=536.0
nSpr=232
visible=true, Entity id=58
x=356.0
y=384.0
nSpr=232
visible=true, Entity id=59
x=364.0
y=384.0
nSpr=230
visible=true, Entity id=60
x=372.0
y=384.0
nSpr=232
visible=true, Entity id=61
x=372.0
y=384.0
nSpr=232
visible=true, Entity id=62
x=372.0
y=392.0
nSpr=231
visible=true, Entity id=63
x=372.0
y=408.0
nSpr=232
visible=true, door4
220.0, 542.0 (15 - bank 5), doorc1a
736.0, 420.0 (16 - bank 5), doorc2a
847.0, 420.0 (15 - bank 5), Perso=new
Coords:(378.0, 41.5 0.0) ==> (378,0)
info=ENEMY
mvt=VIDE, 423.0, 89.5 (80 - bank 0), 378.0, 40.5 (SHADOW), Perso=bigrat1
Coords:(752.0, 477.0 0.0) ==> (731,437)
info=ENEMY
mvt=TOUCHE, 752.0, 487.0 (80 - bank 0), Perso=rat1
Coords:(839.0, 422.0 0.0) ==> (839,418)
info=ENEMY
mvt=VIDE, 839.0, 432.0 (80 - bank 0), 839.0, 421.0 (SHADOW), Perso=rat2
Coords:(882.0, 472.5 0.0) ==> (882,449)
info=ENEMY
mvt=VIDE, 882.0, 482.5 (80 - bank 0), 882.0, 471.5 (SHADOW), 392.0, 320.42676 (SEWER_SMOKE2), 381.85336, 313.56003 (SEWER_VOLUT1), 383.32416, 308.317 (SEWER_VOLUT1), 386.76028, 317.73093 (SEWER_VOLUT1), 400.91928, 313.07513 (SEWER_VOLUT1), 381.63367, 313.9456 (SEWER_VOLUT1), 387.28064, 309.84494 (SEWER_VOLUT1), 391.8032, 317.14572 (SEWER_VOLUT1), 399.15915, 313.70905 (SEWER_VOLUT1), 752.0, 477.0 (DEATH_ANIM1), 752.0, 475.0 (SHADOW_MINUS), 752.0, 477.0 (GOLDCOIN1), 263.0, 303.222 (SEWER_SMOKE1)]
persos=[Perso=Zildo
Coords:(731.75903, 437.00818 0.0)
info=ZILDO
mvt=VIDE, Perso=new
Coords:(378.0, 41.5 0.0) ==> (378,0)
info=ENEMY
mvt=VIDE, Perso=bigrat1
Coords:(752.0, 477.0 0.0) ==> (731,437)
info=ENEMY
mvt=TOUCHE, Perso=rat1
Coords:(839.0, 422.0 0.0) ==> (839,418)
info=ENEMY
mvt=VIDE, Perso=rat2
Coords:(882.0, 472.5 0.0) ==> (882,449)
info=ENEMY
mvt=VIDE]
variables={fishWater=6.0, nettleCount=-1, Carlo=[[MOON, 1], 800, 1], allowedDynamite=yes, Dizzie=[[DYNAMITE, 1], 15, 20], moneyStolen=369.0, BilelIgorVillage=[[EMPTY_BAG,1],100,0]}
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.indexOf(java.lang.String)' on a null object reference
	at zildo.monde.dialog.DialogManagement.continueDialog(DialogManagement.java:193)
	at zildo.monde.dialog.DialogManagement.goOnDialog(DialogManagement.java:225)
	at zildo.server.PlayerManagement.keyPressAttack(PlayerManagement.java:555)
	at zildo.server.PlayerManagement.handleCommon(PlayerManagement.java:174)
	at zildo.server.PlayerManagement.manageKeyboard(PlayerManagement.java:127)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:146)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:193)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1562)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1262)

**Comment by tchegito** [comment]
=> player presses Attack key
=> he's speaking to no one currently
=> a sentence is displayed, which is ready to follow (a sentence containing a '@')

==> but how can it be possible in prison5 ??? No one to speak there !

**Comment by tchegito** [comment]
A complete trace from CrashReporter in prison, just after talking to grandpa:

map=prison (47x28)
sprites=[Perso=Zildo
Coords:(230.0, 256.0 0.0)
info=ZILDO
mvt=VIDE weapon=[MIDSWORD,0] name=Thaden, 231.0, 266.0 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 271.62003, 119.0 (SHADOW), 86.790276, 186.58882 (103 - bank 0), 561.0, 139.0 (BARREL), 720.0, 128.0 (0 - bank 5), igordoor1
176.0, 96.0 (2 - bank 5), gpdoor1
176.0, 208.0 (2 - bank 5), 320.0, 208.0 (2 - bank 5), 320.0, 96.0 (2 - bank 5), 272.0, 96.0 (2 - bank 5), 272.0, 208.0 (2 - bank 5), gpdoor2
128.0, 208.0 (2 - bank 5), igordoor2
128.0, 96.0 (2 - bank 5), 184.0, 280.0 (4 - bank 5), 176.0, 269.0 (5 - bank 5), Entity id=362
x=188.0
y=288.0
nSpr=25
visible=true, Entity id=363
x=200.0
y=288.0
nSpr=26
visible=true, Entity id=364
x=220.0
y=288.0
nSpr=25
visible=true, 268.0, 280.0 (4 - bank 5), Entity id=366
x=244.0
y=288.0
nSpr=25
visible=true, 276.0, 269.0 (5 - bank 5), Entity id=368
x=256.0
y=288.0
nSpr=26
visible=true, Entity id=369
x=268.0
y=288.0
nSpr=25
visible=true, Perso=jaune
Coords:(223.0, 85.0 0.0) ==> (223,60)
info=NEUTRAL
mvt=VIDE, 0.0, 0.0 (80 - bank 0), 0.0, -12.0 (116 - bank 2), 223.0, 85.0 (JAR), Perso=papy
Coords:(103.0, 181.0 0.0)
info=NEUTRAL
mvt=VIDE, 0.0, 0.0 (80 - bank 0), 103.0, 180.0 (SHADOW), Perso=gard2
Coords:(245.0, 260.0 0.0)
info=NEUTRAL
mvt=VIDE, 0.0, 0.0 (80 - bank 0), 0.0, -12.0 (116 - bank 2), 245.0, 260.0 (JAR)]
persos=[Perso=Zildo
Coords:(230.0, 256.0 0.0)
info=ZILDO
mvt=VIDE weapon=[MIDSWORD,0] name=Thaden, Perso=jaune
Coords:(223.0, 85.0 0.0) ==> (223,60)
info=NEUTRAL
mvt=VIDE, Perso=papy
Coords:(103.0, 181.0 0.0)
info=NEUTRAL
mvt=VIDE, Perso=gard2
Coords:(245.0, 260.0 0.0)
info=NEUTRAL
mvt=VIDE]
variables={allowedDynamite=no, Dizzie=[[DYNAMITE, 1], 15, 20], loc:5=1.0, BilelIgorVillage=[[DYNAMITE,1],100,0], loc:4=2.0, fishWater=6.0, nettleCount=-1, Carlo=[[MOON, 1], 800, 1], moneyStolen=460.0}
lastdialog=Lugdunia Prison
�Grandpa�: I don't know how long they will keep me here, but the guards no longer like me at all. There is no kindness left in their eyes.
�Grandpa�: But the guards seemed to think I was too spirited, and locked me down here for public disturbance.
�Grandpa�: I couldn't bear that my little grandaughter had been kidnapped. So since you hadn't returned, I went to the Castle and asked to see the King.
�Grandpa�: Oh my dear, Thaden ! Here you are ! What a pity you must see me in here ...
�Guard�: Don't try anything, boy. I'm watching you !
�Guard�: All right, you can go in.
�Guard�: You can see a prisoner for 10 gold coins.
Lugdunia
�Igor�: With it, you may break through the black guard's armor.
�Igor�: You will see, your sword is now far more powerful than your previous blade.
�Igor�: It is still a little bit hot, but you can have it.
�Igor�: Hello Thaden ! I finished the Sword.
�Thaden�: Hmm... Okay.
�Maltus�: I will follow you. Let's meet on the other side.
�Maltus�: Good luck Thaden ! I ...
�Thaden�: All right.
�Hector�: Unfortunately, it will work against us.
�Hector�: Please be very careful, Thaden. In the prison, I set up many traps to prevent any breakout.
�Thaden�: Very well. I'm going now.
�Maltus�: I'm sure they have hidden the key to access the south side of the prison.
�Hector�: Go for it Thaden. It is time, now, to attack these prison guards.
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.indexOf(java.lang.String)' on a null object reference
	at zildo.monde.dialog.DialogManagement.continueDialog(DialogManagement.java:193)
	at zildo.monde.dialog.DialogManagement.goOnDialog(DialogManagement.java:225)
	at zildo.server.PlayerManagement.keyPressAction(PlayerManagement.java:427)
	at zildo.server.PlayerManagement.handleCommon(PlayerManagement.java:167)
	at zildo.server.PlayerManagement.manageKeyboard(PlayerManagement.java:124)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:146)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:194)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1522)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1239)

**Comment by tchegito** [comment]
Interesting report on CrashReporter (with new fields added, like 'lastHit'):

map=foret (64x64)
sprites=[Perso=Zildo
Coords:(1005.733, 417.49442 0.0)
info=ZILDO
mvt=VIDE pv=3 weapon=[SWORD,0] name=Otto lasthit=last hit by Perso=g1
Coords:(43.344757, 397.43866 0.0) ==> (206,273)
info=ENEMY
mvt=VIDE pv=2 on bosquet, 1006.733, 427.49442 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 108.0, 229.0 (SHADOW), 92.5304, 468.7784 (103 - bank 0), 648.0, 406.0 (STONE_HEAVY), Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, 0.0, 0.0 (80 - bank 0), Perso=new
Coords:(523.0, 661.0 0.0) ==> (10,0)
info=ENEMY
mvt=VIDE pv=2, 523.0, 667.0 (80 - bank 0), 0.0, -12.0 (SHADOW_SMALL), Perso=new
Coords:(724.37976, 804.62024 0.0) ==> (716,834)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 724.37976, 803.62024 (SHADOW), Perso=new
Coords:(357.54523, 749.227 0.0) ==> (314,702)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 357.54523, 748.227 (SHADOW), Perso=new
Coords:(475.70013, 341.70013 0.0) ==> (456,323)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 475.70013, 340.70013 (SHADOW), Perso=new
Coords:(498.0, 198.0 0.0)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 498.0, 197.0 (SHADOW)]
persos=[Perso=Zildo
Coords:(1005.733, 417.49442 0.0)
info=ZILDO
mvt=VIDE pv=3 weapon=[SWORD,0] name=Otto lasthit=last hit by Perso=g1
Coords:(43.344757, 397.43866 0.0) ==> (206,273)
info=ENEMY
mvt=VIDE pv=2 on bosquet, Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, Perso=new
Coords:(523.0, 661.0 0.0) ==> (10,0)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(724.37976, 804.62024 0.0) ==> (716,834)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(357.54523, 749.227 0.0) ==> (314,702)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(475.70013, 341.70013 0.0) ==> (456,323)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(498.0, 198.0 0.0)
info=ENEMY
mvt=VIDE pv=2]
variables={nettleCount=-1, Carlo=[[MOON, 1], 800, 1], allowedDynamite=yes, Dizzie=[[DYNAMITE, 1], 15, 20], BilelIgorVillage=[[EMPTY_BAG, 1], 100, 2], [[BLUEDROP, 1], 15, -1], [[DYNAMITE, 1], 100, 2]}
scripts=1 scripts running {[@scene@bosquet_still_gard[0 on [speak maltus bosquet.maltus.2 , markQuest bosquet_still_gard ]],}
lastdialog=Lugdunia Forest
�Maltus�: What is it, Otto ? What are you waiting for ? Clear the way !
Polaky
�Otto�: I'd rather return to see Maltus.
Lugdunia Forest
�Eric�: I've always wondered what's inside, but I'm too scared to set foot near there.
�Eric�: Hi ! Did you see the house on the hill ? It has been abandoned for quite a while !
�Eric�: Do you want to try to get in the house ?
�Eric�: I've always wondered what's inside, but I'm too scared to set foot near there.
�Eric�: Hi ! Did you see the house on the hill ? It has been abandoned for quite a while !
Polaky
�Otto�: I'd rather return to see Maltus.
Lugdunia Forest
�Maltus�: What is it, Otto ? What are you waiting for ? Clear the way !
�Maltus�: There are guards patrolling here. Otto, defeat them and come back as soon as the way is clear.
�Maltus�: This way.
�Maltus�: Follow me, we are close to Polaky.
�Otto�: Fortunately, my Sword came in handy. So, where do we go now ?
�Maltus�: Run away, me ? Absolutely not ! I was scouting ahead ... to help accomplish our mission !
�Otto�: You would know if you hadn't run away !
�Maltus�: Ah Otto ! Where have you been ?
�Otto�: Maltus !!!
�Maltus�: What's going on ?
�Maltus�: We can't get passed this log.
�Otto�: Very well, let's go !
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.indexOf(java.lang.String)' on a null object reference
	at zildo.monde.dialog.DialogManagement.continueDialog(DialogManagement.java:193)
	at zildo.monde.dialog.DialogManagement.goOnDialog(DialogManagement.java:225)
	at zildo.server.PlayerManagement.keyPressAction(PlayerManagement.java:427)
	at zildo.server.PlayerManagement.handleCommon(PlayerManagement.java:167)
	at zildo.server.PlayerManagement.manageKeyboard(PlayerManagement.java:127)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:153)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:194)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1531)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1248)


```

**Comment by tchegito** [comment]
[Fixed in 2.44] 

TU added

1d6eb07e04e617d8e0a16d1ed43e79a9d9dae13c
